cfme/common/provider_views.py

```diff
@@ -314,6 +314,7 @@ def is_displayed(self):
     sidebar = View.nested(ProviderSideBar)
     search = View.nested(Search)
     including_entities = View.include(ProviderEntitiesView, use_parent=True)
+    add_button = Text('//*[@id="main_div"]//div/a')
```

Comment:
Q: Is there a reason that `Button` can't be used here?

Reply:
Oh no! I forgot to change that to `Button`. Will do that right away!

Reply:
@john-dupuy I think I might now have a reason to use `Text` instead of `Button`. I cannot use locator/xpath of the element with `Button`, and using xpath is a sure-shot way to access the element. I think I'll stick to `Text`.

Reply:
@valaparthvi Ah I see, I'm surprised it doesn't work. You should be able to do `Button("Add a Provider")` and the locator should take care of the rest. 

```python,
    # TODO: Handle input value the same way as text for other tags
    def __locator__(self):
        return (
            './/*[(self::a or self::button or (self::input and (@type="button" or @type="submit")))'
            ' and contains(@class, "btn") {}]'.format(self.locator_conditions))
```

Reply:
Let me try and see

Reply:
Yep, just was able to confirm, with `Button` you pass the text of the button, not the xpath, so using `Button("Add a Provider")` will work. The `__init__` function of `Button` will build the xpath for you. 

https://github.com/RedHatQE/widgetastic.patternfly/blob/cf5f288254dca7f5ae00f17a50cd377119198c6a/src/widgetastic_patternfly/__init__.py#L125-L143

Reply:
@john-dupuy It worked like magic!